drivers/counter/counter_xlnx_axi_timer.c

```diff
@@ -356,7 +357,7 @@ static const struct counter_driver_api xlnx_axi_timer_driver_api = {
 	{								\
 		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority),	\
 			    xlnx_axi_timer_isr,				\
-			    DEVICE_GET(xlnx_axi_timer_##n), 0);		\
+			    DEVICE_DT_INST_GET(n), 0);			\
```

Comment:
This pattern is showing up often. We could write new helpers at some point to cut boilerplate:

```c
#define IRQ_CONNECT_DT(node_id, isr, flags) \
        IRQ_CONNECT(DT_IRQN(node_id), DT_IRQ(node_id, priority), \
                    isr, DEVICE_DT_GET(node_id), flags)
#define IRQ_CONNECT_DT_INST(inst, isr, flags) \
        IRQ_CONNECT_DT(DT_DRV_INST(inst), isr, flags)
```

Not for this PR, though. Just thinking out loud.

Reply:
Yeah, I've noticed the same pattern myself. :)